From ccceaf91ef88b15f9289a12184b57eb2b4fcb438 Mon Sep 17 00:00:00 2001
From: Rahul Lakkireddy <rahul.lakkireddy@chelsio.com>
Date: Mon, 9 Jul 2018 12:57:53 +0530
Subject: [PATCH 105/107] cxgb4: fix missing break in switch and indent return
 statements

commit 5dc874252faa818426480a7c00fa05738fe05402
Author: Colin Ian King <colin.king@canonical.com>
Date:   Fri Oct 13 17:29:00 2017 +0100

    cxgb4: fix missing break in switch and indent return statements

    The break statement for the Macronix case is missing and will
    fall through to the Winbond case and re-assign the size setting.
    Fix this by adding the missing break statement.  Also correctly
    indent the return statements.

    Detected by CoverityScan, CID#1458020 ("Missing break in switch")

    Fixes: 96ac18f14a5a ("cxgb4: Add support for new flash parts")
    Signed-off-by: Colin Ian King <colin.king@canonical.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>
---
 drivers/net/ethernet/chelsio/cxgb4/t4_hw.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/chelsio/cxgb4/t4_hw.c b/drivers/net/ethernet/chelsio/cxgb4/t4_hw.c
index 4478bb37..481cc3ec 100644
--- a/drivers/net/ethernet/chelsio/cxgb4/t4_hw.c
+++ b/drivers/net/ethernet/chelsio/cxgb4/t4_hw.c
@@ -8057,7 +8057,7 @@ static int t4_get_flash_params(struct adapter *adap)
 		default:
 			dev_err(adap->pdev_dev, "Micron Flash Part has bad size, ID = %#x, Density code = %#x\n",
 				flashid, density);
-		return -EINVAL;
+			return -EINVAL;
 		}
 		break;
 	}
@@ -8076,8 +8076,9 @@ static int t4_get_flash_params(struct adapter *adap)
 		default:
 			dev_err(adap->pdev_dev, "Macronix Flash Part has bad size, ID = %#x, Density code = %#x\n",
 				flashid, density);
-		return -EINVAL;
+			return -EINVAL;
 		}
+		break;
 	}
 	case 0xef: { /* Winbond */
 		/* This Density -> Size decoding table is taken from Winbond
@@ -8094,7 +8095,7 @@ static int t4_get_flash_params(struct adapter *adap)
 		default:
 			dev_err(adap->pdev_dev, "Winbond Flash Part has bad size, ID = %#x, Density code = %#x\n",
 				flashid, density);
-		return -EINVAL;
+			return -EINVAL;
 		}
 		break;
 	}
-- 
2.14.1

